stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -633,7 +675,10 @@ public class ChatClient internal constructor(
     }
 
     private fun getTokenAndConnect(connect: () -> Unit) {
-        config.tokenManager.loadAsync {
+        tokenManager.loadAsync {
+            if (it.isSuccess) {
+                state.token = it.data()
```

Comment:
Instead of store the token into the `state` object, Why don't we use hte token into the `TokenManager`?
At the end it could be not updated, because the token manager could be asked to reload the token from multiple part of our code